firebase-firestore/src/main/java/com/google/firebase/firestore/remote/WatchChangeAggregatorTestingHooks.java

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.remote;
+
+import static com.google.firebase.firestore.util.Preconditions.checkNotNull;
+
+import androidx.annotation.AnyThread;
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import com.google.auto.value.AutoValue;
+import com.google.firebase.firestore.ListenerRegistration;
+import com.google.firebase.firestore.util.Executors;
+import java.util.HashMap;
+import java.util.Map;
+
+final class WatchChangeAggregatorTestingHooks {
+
+  private WatchChangeAggregatorTestingHooks() {}
```

Comment:
in web, the TestingHooks class used singleton design. Here, I assume, it is done by making the constructor private and accessing the existenceFilterMismatchListeners via WatchChangeAggregatorTestingHooksAccessor. Why we want to use different designs in 2 SDKs?

Reply:
We definitely _could_ use a singleton "Testing Hooks" class here in Android as well. Here in Android I chose to use `static` methods rather than expose a singleton instance, which are effectively the same thing. I felt like using `static` methods in Java was more idiomatic. I didn't think that consistency of the design of testing hooks across the SDKs was terribly important because each language has its own way of doing that sort of thing.

All that being said, I'll see what I can do to make it more consistent with the Web SDK and see what you think. Maybe it will be better? I honestly didn't try.